### PR TITLE
Upgrade NUnit version

### LINKED ISSUE #14: Update NUnit version
Currently checks for a reference to 3.2.0

### PR DESCRIPTION
**Pull request by ChrisMaddock**
And finally, update the NUnit version to 3.4.1. (Fixes #14) Apologies for all the PRs, hoped splitting them up would make it easier to review.

Is it worth doing another release of this? These three PRs are all aesthetic, but it would be good to have #7 in a release. The currently released version is also using NUnit 3.0.1.

### CODE REVIEW

**Comment by rprouse** [comment]
Should we bump the version in the `vsixmanifest` at the same time?

**Comment by ChrisMaddock** [comment]
@rprouse - I presumed that would just be done on release? (I wasn't intending this PR to be a release - just seemed sensible to mention it here rather than a new issue!)

**Comment by CharliePoole** [comment]
@ChrisMaddock So what does this actually do? Doesn't it generate tests the same way whatever the NUnit version?

**Comment by ChrisMaddock** [comment]
@CharliePoole - one of bits of options is to generate a new Unit Test project. This is created with a reference/nuget packages.config pointing at whichever version is specified.

**Comment by CharliePoole** [comment]
And the user doesn't get a choice? Except between major versions?

**Comment by CharliePoole** [comment]
If we always want to use the latest NUnit 3 release, why code the version at all?

**Comment by ChrisMaddock** [comment]
The user can change the reference as they wish, but this is the default version that a new project will be created with. 

As far as I can see (though trial and error - if there are any docs - I couldn't find them!)  - this won't ever overwrite anything the user sets (even if the nunit reference added is _lower_ than the specified version).

> If we always want to use the latest NUnit 3 release, why code the version at all?

I thought the same, but if it's possible, I couldn't work out how! A version string is required - I don't know if it's possible to pass a `*` or something to 'default to latest'. @OsirisTerje, any ideas?

**Comment by OsirisTerje** [comment]
The test generator _extension_ as it is now is bound to a specific version at creation time.  What the user do afterwards is of course up to the user.  All we do is to give parameters to an internal method on  the test platform, and that method seems to require a version number.  But I will ask the MS guys if it will work without anything here (or  a `*` as you say.   so that it always use the latest, which it should do for NUnit3.  For NUnit2 we need to bind it to 2.6.4, otherwise it will step up to latest NUnit3.   The PR is good otherwise,we can just merge this in.

**Comment by OsirisTerje** [comment]
I've got a answer from the MS PG that if we pass in a null for the version, it will use the latest.  Can you do that change for the NUnit 3  @ChrisMaddock ?

**Comment by ChrisMaddock** [comment]
Tried it out and it seems to work @OsirisTerje - done!

File: TestGeneration.Extensions.NUnit/source.extension.vsixmanifest [hunk 1]
```diff
@@ -7,8 +7,8 @@
 Creates Unit tests and Intellitests with both NUnit 2.6.4 and NUnit 3 frameworks.
 </Description>
     <MoreInfo>https://github.com/nunit/nunit-vs-testgenerator</MoreInfo>
-    <GettingStartedGuide>https://github.com/nunit/docs/wiki/Visual-Studio-Test-Generator</GettingStartedGuide>
     <License>license.txt</License>
+    <GettingStartedGuide>https://github.com/nunit/docs/wiki/Visual-Studio-Test-Generator</GettingStartedGuide>
```
Comment (ChrisMaddock) [inline TestGeneration.Extensions.NUnit/source.extension.vsixmanifest:11]: I couldn't build without re-ording the vsixmanifest elements like this. I added the <GettingStartedGuide> tag in another PR, but I didn't think there was any required ordering in vsixmanifests. Do you know if there is?

Comment (OsirisTerje) [inline TestGeneration.Extensions.NUnit/source.extension.vsixmanifest:11]: There should not be, afaik, but... anyway...  if it works it works.  I'll merge this now.

Comment (ChrisMaddock) [inline TestGeneration.Extensions.NUnit/source.extension.vsixmanifest:11]: Bizarre! Thanks!

**Pull request merged**